### ISSUE TITLE
Issue with failing t/cert_formats.t for OpenSSL 3.0.0

### ISSUE THREAD

**Issue by michal-josef-spacek** [noxxi/p5-io-socket-ssl#107]
There are failing tests in t/cert_formats.t for OpenSSL 3.0.0
```
[skim@man p5-io-socket-ssl]$ perl t/cert_formats.t
1..6
ok 1 - [PEM] ssl connect success
ok 2 - [PEM_one_file] ssl connect success
ok 3 - [PEM_keyenc] ssl connect success
ok 4 - [DER] ssl connect success
# start_SSL failed: Failed to load certificate from file certs/server.p12: [format:PEM] error:0480006C:PEM routines::no start line error:0A080009:SSL routines::PEM lib ** [format:DER] error:068000A8:asn1 encoding routines::wrong tag error:0688010A:asn1 encoding routines::nested asn1 error error:0688010A:asn1 encoding routines::nested asn1 error error:0A08000D:SSL routines::ASN1 lib ** [format:PKCS12] error:0308010C:digital envelope routines::unsupported **
not ok 5 - [PKCS12] ssl connect failed: SSL connect attempt failed
#   Failed test '[PKCS12] ssl connect failed: SSL connect attempt failed'
#   at t/cert_formats.t line 67.
# start_SSL failed: Failed to load certificate from file certs/server_enc.p12: [format:PEM] error:0480006C:PEM routines::no start line error:0A080009:SSL routines::PEM lib ** [format:DER] error:068000A8:asn1 encoding routines::wrong tag error:0688010A:asn1 encoding routines::nested asn1 error error:0688010A:asn1 encoding routines::nested asn1 error error:0A08000D:SSL routines::ASN1 lib ** [format:PKCS12] error:11800071:PKCS12 routines::mac verify failure error:0308010C:digital envelope routines::unsupported **
not ok 6 - [PKCS12_enc] ssl connect failed: SSL connect attempt failed
#   Failed test '[PKCS12_enc] ssl connect failed: SSL connect attempt failed'
#   at t/cert_formats.t line 67.
# Looks like you failed 2 tests of 6.
```

Issue is with unsupported algorithms.

**Comment by michal-josef-spacek** [comment]
I prepared PR with fix for OpenSSL 3.0.0. - #108

**Comment by noxxi** [comment]
fixed in b410f80

**Comment by michal-josef-spacek** [comment]
@noxxi Thank you :-)
